mesos-client/src/test/scala/com/mesosphere/mesos/client/SessionActorTest.scala

```diff
@@ -0,0 +1,38 @@
+package com.mesosphere.mesos.client
+import akka.http.scaladsl.model.{HttpEntity, HttpRequest, HttpResponse}
+import akka.http.scaladsl.model.headers.HttpCredentials
+import akka.testkit.TestProbe
+import com.mesosphere.utils.AkkaUnitTest
+
+import scala.concurrent.Future
+
+class SessionActorTest extends AkkaUnitTest {
+
+  "The Session Actor" should {
+    "replied with a response to the original sender" in {
```

Comment:
Only one test? I understand that mocking IAM responses might be difficult but we need more tests. And maybe we don't need a complicated mock (like `app_mock.py`), a simple akka-http server [like this](https://github.com/mesosphere/marathon/blob/master/src/test/scala/mesosphere/marathon/core/health/impl/HealthCheckWorkerTest.scala#L59) might already suffice.

Reply:
Yeah, I would really like to see an integration test which uses a mock IAM service, testing the following scenarios:

* Happy path (authentication works)
* semi-happy path (authentication fails initially because the token is invalid), a new token is fetched and the call in question is resubmitted.